src/components/Collapse.tsx

```diff
@@ -8,12 +8,15 @@ interface Props {
   isOpen?: boolean;
   label: ReactNode;
   onClick?: (open: boolean) => void;
+  'data-fs-element'?: string;
 }
 
 export const Collapse = ({ className, isOpen, label, children, onClick, ...props }: PropsWithChildren<Props>) => {
   const [isOpenState, setIsOpenState] = React.useState(isOpen);
   const styles = useStyles2(getStyles);
 
+  const analyticsName = props['data-fs-element'];
+  delete props['data-fs-element'];
```

Comment:
You can do this a cleaner way when you have kebab-case props by using the renaming function when destructuring:

```
{ className, isOpen, label, children, onClick, 'data-fs-element': dataFsElement, ...props }
...
<button data-fs-element={dataFsElement} ... />
```

Then you get rid of all the horrible props mutation 😃

Reply:
haha all the cool kids are doing prop mutation